File: Chapter08/src/main/java/chapter8/ThreadsExample.java
package chapter8;

public class ThreadsExample {

    public static void main(String[] args) {
        Thread t1 = new Thread(() -> {
            for (int i = 0; i < 100; i++) {
                System.out.println("T1: " + i);
            }
        });
        t1.setDaemon(true);
        t1.start();
        

        new Thread(() -> {
            for (int i = 0; i < 100; i++) {
                System.out.println("T2: " + i);
            }
        }).start();

    }
}

File: Chapter08/src/main/kotlin/chapter8/ThreadsExample.kt
package chapter8

import kotlin.concurrent.thread

fun main(args: Array<String>) {

    val t1 = thread {
        for (i in 1..100) {
            println("T1: $i")
        }
    }

    val t2 = thread(start = false) {
        for (i in 1..100) {
            println("T2: $i")
        }
    }

    t2.start()

    val t3 = thread(isDaemon = true) {
        for (i in 1..1_000_000) {
            println("T3: $i")
        }
    }
}